decisiontree/models.py

```diff
@@ -114,141 +114,156 @@ def helper_text(self):
+        TreeState, blank=True, null=True,
+        help_text="None if the session is complete.")
+    num_tries = models.PositiveIntegerField(
+        help_text="The number of times the user has tried to answer the "
+                  "current question.")
     # this flag stores the difference between completed
     # on its own, or manually canceled.
-    canceled = models.NullBooleanField(blank=True, null=True) 
+    canceled = models.NullBooleanField(blank=True, null=True)
     last_modified = models.DateTimeField(auto_now=True, null=True)
-     
-    def __unicode__(self):
-        if self.state:
-            text = self.state
-        else:
-            text = "completed"
-        return u"%s : %s" % (self.connection.identity, text)
 
+    def __str__(self):
+        state = self.state or "completed"
+        return u"%s : %s" % (self.connection.identity, state)
 
+
+@python_2_unicode_compatible
 class Entry(models.Model):
-    """ An Entry is a single successful movement within
-        a Session.  It represents an accepted Transition 
-        from one state to another within the tree. """ 
+    """
+    An Entry is a single successful movement within a Session.  It represents
+    an accepted Transition from one state to another within the tree.
+    """
     session = models.ForeignKey(Session, related_name='entries')
     sequence_id = models.IntegerField()
     transition = models.ForeignKey(Transition, related_name='entries')
     time = models.DateTimeField(auto_now_add=True, db_index=True)
-    text = models.CharField(max_length=160)    
+    text = models.CharField(max_length=160)
     tags = models.ManyToManyField('Tag', related_name='entries')
-    
-    def __unicode__(self):
-        return u"%s-%s: %s - %s" % (self.session.id, self.sequence_id, 
-                self.transition.current_state.question, self.text)
-    
-    def meta_data(self):
-        return "%s - %s %s" % (
-            self.session.person.phone,
-            self.time.strftime("%a %b %e"),
-            self.time.strftime("%I:%M %p"))
-    
+
+    class Meta(object):
+        verbose_name_plural = "Entries"
+        ordering = ('sequence_id',)
+
+    def __str__(self):
+        return u"%s-%s: %s - %s" % (
+            self.session.pk, self.sequence_id,
+            self.transition.current_state.question, self.text)
+
     def display_text(self):
         # assume that the display text is just the text,
         # since this is what it is for free text entries
         return self.text
 
-    class Meta:
-        verbose_name_plural="Entries"
-        ordering = ('sequence_id',)
+    def meta_data(self):
+        return "%s - %s %s" % (
+            self.session.person.phone,
+            self.time.strftime("%a %b %e"),
+            self.time.strftime("%I:%M %p"))
 
 
+@python_2_unicode_compatible
 class Tag(models.Model):
     name = models.CharField(unique=True, max_length=100)
     recipients = models.ManyToManyField(User, related_name='tags')
```

Comment:
Again, apologies for not noting this the first time around, but we should probably use `settings.AUTH_USER_MODEL` here and `get_user_model()` in forms.py.